### PR TITLE
Implement basic version of web socket server #19

### PR DESCRIPTION
**Pull request by Vicx95**
Create initial CMakeList

Implement basic version of chat server

### CODE REVIEW

File: server/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+cmake_minimum_required(VERSION 3.14)
```
Comment (lukasz126) [inline server/CMakeLists.txt:1]: maybe version 3.5? like in rest cmakelists?

File: server/chatserver.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ChatServer::ChatServer(quint16 port, QObject *parent)
+    : QObject{parent},
+      m_webSocketServer{new QWebSocketServer("Chat Server",
+                            QWebSocketServer::NonSecureMode,
+                            this)}
+{
+    if(m_webSocketServer->listen(QHostAddress::Any, port))
+    {
+        qInfo() << "Server listening on port " << port;
+        connect(&*m_webSocketServer, &QWebSocketServer::newConnection, this, &ChatServer::onNewConnection);
```
Comment (lukasz126) [inline server/chatserver.cpp:18]: why &*m_webSocketServer instead of m_webSocketServer?

File: server/CMakeLists.txt (hunk 1 above)
Comment (Vicx95) [reply server/CMakeLists.txt:1]: ok I will change it

File: server/chatserver.cpp (hunk 2 above)
Comment (Vicx95) [reply server/chatserver.cpp:18]: this was left after trying to use smart pointers, I will change it

**Review by lukasz126 — APPROVED** [review]

**Comment by ziobron** [comment]
Your PR was merged!
🏅 2 XP granted. Thanks @Vicx95 for making course materials better!

**Pull request merged**